workers/unity/Packages/io.improbable.worker.sdk.mobile/BuildPostProcessXCode.cs

```diff
@@ -0,0 +1,119 @@
+                return;
+            }
+
+            var xcodeObject = Activator.CreateInstance(pbxType);
+
+            // Instantiate PBXProject and read from xcode project file.
+            var projPath = InvokeStaticMethod<string>("GetPBXProjectPath", path);
+            var xcodeText = File.ReadAllText(projPath);
+            InvokeMethod(xcodeObject, "ReadFromString", xcodeText);
+
+            // Get Target GUIDs
+            var unityTargetName = InvokeStaticMethod<string>("GetUnityTargetName");
+            var unityTestTargetName = InvokeStaticMethod<string>("GetUnityTestTargetName");
+            var targetGUID = InvokeMethod<string>(xcodeObject, "TargetGuidByName", unityTargetName);
+            var targetTestingGUID = InvokeMethod<string>(xcodeObject, "TargetGuidByName", unityTestTargetName);
+
+            // Enumerate configGUIDs that need library path patching
+            var configNames = InvokeMethod<IEnumerable<string>>(xcodeObject, "BuildConfigNames");
+            var configGUIDs = configNames
+                .Select(configName => InvokeMethod<string>(xcodeObject, "BuildConfigByName", targetGUID, configName))
+                .Concat(configNames.Select(configName =>
+                    InvokeMethod<string>(xcodeObject, "BuildConfigByName", targetTestingGUID, configName)));
+
+            // Update library paths for each config
+            foreach (var configGUID in configGUIDs)
+            {
+                InvokeMethod<string>(xcodeObject, "UpdateBuildPropertyForConfig", configGUID, "LIBRARY_SEARCH_PATHS",
+                    null, new[]
+                    {
+                        "$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS/arm",
+                        "$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS/x86_64"
+                    });
+
+                // Add platform specific paths
+                InvokeMethod<string>(xcodeObject, "UpdateBuildPropertyForConfig", configGUID,
+                    "LIBRARY_SEARCH_PATHS[sdk=iphoneos*]", new[]
+                    {
+                        "$(inherited)",
+                        "$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS/arm"
+                    }, null);
+
+                InvokeMethod<string>(xcodeObject, "UpdateBuildPropertyForConfig", configGUID,
+                    "LIBRARY_SEARCH_PATHS[sdk=iphonesimulator*]",
+                    new[]
+                    {
+                        "$(inherited)",
+                        "$(SRCROOT)/Libraries/io.improbable.worker.sdk.mobile/Plugins/Improbable/Core/iOS/x86_64"
+                    }, null);
+            }
+
+            // Save changes to xcode project file
+            InvokeMethod<string>(xcodeObject, "WriteToFile", projPath);
+        }
+
+        private static T InvokeStaticMethod<T>(string methodName, params object[] parameters)
```

Comment:
that could be factored out into something more generic, but doesn't need to happen now